Discussion draft of field requirements.

hughstimson opened the pull request:
This adds field requirements and documentation for four linked datasets comprising the standard, in particular the core "consultations" dataset. This is a draft for discussion.

Sample data is also added demonstrating what real world data might look like when publishing to the standard.

hughstimson committed:
Adds draft requirements documentation.

This adds field requirements and documentation for four linked datasets
comprising the standard, in particular the core "consultations" dataset.
This is a draft for discussion.

hughstimson committed:
Standardizes capitalization of ConsultationId.

hughstimson committed:
Updates README.md to reflect added requirements.

hughstimson committed:
Adds sample data.

"Sample Data is provided in the `SampleData` folder demonstrating how
real-world data (derived from EngageBC and PlaceSpeak.com) could be
formatted to fit the standard. Note that not all fields are required, so
not all data being published to the standard will have a structure
identical to this sample."

jpmckinney commented on Optional-public-events-dataset-requirements.md:
```diff
@@ -0,0 +1,23 @@
+PublicEventAddress1    | TEXT              | The street address of the public event, if applicable, or the first line of the address for multi-line addresses.
+PublicEventAddress2    | TEXT              | The second line of the address for multi-line addresses.
+PublicEventCity        | TEXT              | The city of the address.
+PublicEventState       | TEXT              | The province or state of the address.
```
Use the more generic terms "locality" and "region" instead of "city" and "state", which are factually incorrect for many places in Canada and the rest of the world. Schema.org, vCard and many other standards use those generic terms.

hughstimson committed:
Changes "State"->"Region", "City"->"Locality".

Some feedback from @jpmckinney:
https://github.com/bcgov/openconsult/pull/2/files/173860f49bdf2b8af20525fdb0d0338b81064de6#r59150325

>Use the more generic terms "locality" and "region" instead of "city"
and "state", which are factually incorrect for many places in Canada and
the rest of the world. Schema.org, vCard and many other standards use
those generic terms.

I'm used to trying to stick to US conventions but no need to do so here.
Updates made in the requirements and sample data.

hughstimson committed:
Attempts to clarify OrganizationName.

hughstimson committed:
Adds reference to BC geographical names service.

hughstimson committed:
Changes StatusCurrentMapped to StatusCurrentStandardized.

Hopefully that's a little clearer.

hughstimson committed:
Tweaks some whitespace.

hughstimson committed:
Merge branch 'master' into requirements-working

Conflicts:
	README.md
